src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessNextDataSetVersionMappingsOrchestrationTests.cs

```diff
@@ -0,0 +1,92 @@
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Model;
+using Microsoft.DurableTask;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using static GovUk.Education.ExploreEducationStatistics.Common.Tests.Utils.MockUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests.Functions;
+
+public abstract class ProcessNextDataSetVersionMappingsOrchestrationTests
+{
+    public class ProcessNextDataSetVersionMappingsTests
+    {
+        [Fact]
+        public async Task Success()
+        {
+            var mockOrchestrationContext = DefaultMockOrchestrationContext();
+            var activitySequence = new MockSequence();
+
+            string[] expectedActivitySequence =
+            [
+                ActivityNames.CopyCsvFiles,
+                ActivityNames.CreateMappings,
+                ActivityNames.ApplyAutoMappings,
+                ActivityNames.CompleteNextDataSetVersionMappingProcessing,
+            ];
+
+            foreach (var activityName in expectedActivitySequence)
+            {
+                mockOrchestrationContext
+                    .InSequence(activitySequence)
+                    .Setup(context => context.CallActivityAsync(activityName,
+                        mockOrchestrationContext.Object.InstanceId,
+                        null))
+                    .Returns(Task.CompletedTask);
+            }
+
+            await ProcessNextDataSetVersion(mockOrchestrationContext.Object);
+
+            VerifyAllMocks(mockOrchestrationContext);
+        }
+
+        [Fact]
+        public async Task ActivityFunctionThrowsException_CallsHandleFailureActivity()
+        {
+            var mockOrchestrationContext = DefaultMockOrchestrationContext();
+
+            var activitySequence = new MockSequence();
+
+            mockOrchestrationContext
+                .InSequence(activitySequence)
+                .Setup(context =>
+                    context.CallActivityAsync(ActivityNames.CopyCsvFiles,
+                        mockOrchestrationContext.Object.InstanceId,
+                        null))
+                .Throws<Exception>();
+
+            mockOrchestrationContext
+                .InSequence(activitySequence)
+                .Setup(context =>
+                    context.CallActivityAsync(ActivityNames.HandleProcessingFailure,
+                        mockOrchestrationContext.Object.InstanceId,
+                        null))
+                .Returns(Task.CompletedTask);
+
+            await ProcessNextDataSetVersion(mockOrchestrationContext.Object);
+
+            VerifyAllMocks(mockOrchestrationContext);
+        }
+
+        private async Task ProcessNextDataSetVersion(TaskOrchestrationContext orchestrationContext)
```

Comment:
I think the name of this method now should be `ProcessNextDataSetVersionMappings`.

Reply:
Thanks, updated!